--- a/arch/arm/boot/dts/suniv-f1c100s.dtsi	2020-01-27 07:23:03.000000000 +0700
+++ b/arch/arm/boot/dts/suniv-f1c100s.dtsi	2022-10-15 20:49:21.190314130 +0700
@@ -89,6 +89,11 @@
 			#interrupt-cells = <3>;
 			#gpio-cells = <3>;
 
+			ir_pin: ir-pin {
+				pins = "PE11";
+				function = "ir";
+			};
+
 			uart0_pe_pins: uart0-pe-pins {
 				pins = "PE0", "PE1";
 				function = "uart0";
@@ -102,6 +107,17 @@
 			clocks = <&osc24M>;
 		};
 
+		cir: cir@1c22c00 {
+			compatible = "allwinner,sun6i-a31-ir";
+			clocks = <&ccu CLK_BUS_IR>, 
+						<&ccu CLK_CIR>;
+			clock-names = "apb", "ir";
+			resets = <&ccu RST_BUS_IR>;
+			interrupts = <6>;
+			reg = <0x01c22c00 0x40>;
+			status = "disabled";
+		};
+
 		wdt: watchdog@1c20ca0 {
 			compatible = "allwinner,suniv-f1c100s-wdt",
 				     "allwinner,sun4i-a10-wdt";
