--- a/arch/arm/boot/dts/suniv-f1c100s.dtsi	2022-12-11 16:14:08.135398163 +0700
+++ b/arch/arm/boot/dts/suniv-f1c100s.dtsi	2023-02-01 05:18:54.080198393 +0700
@@ -3,10 +3,11 @@
  * Copyright 2018 Icenowy Zheng <icenowy@aosc.io>
  * Copyright 2018 Mesih Kilinc <mesihkilinc@gmail.com>
  */
-
+#include <dt-bindings/interrupt-controller/irq.h>
 #include <dt-bindings/clock/suniv-ccu-f1c100s.h>
 #include <dt-bindings/reset/suniv-ccu-f1c100s.h>
 #include <dt-bindings/dma/sun4i-a10.h>
+#include <dt-bindings/pinctrl/sun4i-a10.h>
 
 / {
 	#address-cells = <1>;
@@ -190,6 +191,38 @@
 				pins = "PF0", "PF1", "PF2", "PF3", "PF4", "PF5";
 				function = "mmc0";
 			};
+
+			i2c0_pe_pins: i2c0-pe-pins {
+				pins = "PE11", "PE12";
+				function = "i2c0";
+			};
+
+			i2c0_pd_pins: i2c0-pd-pins {
+				pins = "PD0", "PD12";
+				function = "i2c0";
+				bias-pull-up;
+			};
+
+			i2c1_pd_pins: i2c1-pd-pins {
+				pins = "PD5", "PD6";
+				function = "i2c1";
+			};
+
+			i2c2_pd_pins: i2c2-pd-pins {
+				pins = "PD15", "PD16";
+				function = "i2c2";
+				bias-pull-up;
+			};
+
+			i2c2_pe_pins: i2c2-pe-pins {
+				pins = "PE0", "PE1";
+				function = "i2c2";
+			};
+
+			i2s0_pd_pins: i2s0-pd-pins {
+				pins = "PD7", "PD8", "PD9", /*"PD10",*/ "PE6";
+				function = "i2s";
+			};
 		};
 
 		timer@1c20c00 {
@@ -216,6 +249,23 @@
 			reg = <0x01c20ca0 0x20>;
 		};
 
+		i2s0: i2s@1c22000 {
+			#sound-dai-cells = <0>;
+			compatible = "allwinner,sun6i-a31-i2s";
+			reg = <0x01c22000 0x400>;
+			interrupts = <35>;
+			clocks = <&ccu CLK_BUS_I2S0>, <&ccu CLK_I2S>;
+			clock-names = "apb", "mod";
+			resets = <&ccu RST_BUS_I2S0>;
+			dmas = <&dma SUN4I_DMA_NORMAL 0x0e>,
+				   <&dma SUN4I_DMA_NORMAL 0x0e>;
+			dma-names = "rx", "tx";
+			pinctrl-names = "default";
+			pinctrl-0 = <&i2s0_pd_pins>;
+			
+			status = "disabled";
+		};
+
 		uart0: serial@1c25000 {
 			compatible = "snps,dw-apb-uart";
 			reg = <0x01c25000 0x400>;
@@ -249,6 +299,41 @@
 			status = "disabled";
 		};
 
+		i2c0: i2c@1c27000 {
+			compatible = "allwinner,sun6i-a31-i2c";
+			reg = <0x01c27000 0x400>;
+			interrupts = <7>;
+			clocks = <&ccu CLK_BUS_I2C0>;
+			resets = <&ccu RST_BUS_I2C0>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&i2c0_pd_pins>;
+			status = "disabled";
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+
+		i2c1: i2c@1c27400 {
+			compatible = "allwinner,sun6i-a31-i2c";
+			reg = <0x01c27400 0x400>;
+			interrupts = <8>;
+			clocks = <&ccu CLK_BUS_I2C1>;
+			resets = <&ccu RST_BUS_I2C1>;
+			status = "disabled";
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+
+		i2c2: i2c@1c27800 {
+			compatible = "allwinner,sun6i-a31-i2c";
+			reg = <0x01c27800 0x400>;
+			interrupts = <9>;
+			clocks = <&ccu CLK_BUS_I2C2>;
+			resets = <&ccu RST_BUS_I2C2>;
+			status = "disabled";
+			#address-cells = <1>;
+			#size-cells = <0>;
+		};
+
 		usb_otg: usb@1c13000 {
 			compatible = "allwinner,suniv-f1c100s-musb";
 			reg = <0x01c13000 0x0400>;
